docs/enhancements/4-nodes-cluster-deployment.md

```diff
@@ -0,0 +1,154 @@
+
+- Modify Assisted Service's manifests logic to allow for partial manifests to be provided. Openshift
+installer applies the provided manifests in sequence. If 2 manifests for the same CR are provided,
+then the one applied last will be the one used. This is a problem when the user only wants to modify
+1 field. Here's an [example from a PoC](https://github.com/flaper87/assisted-service/commit/c6d72827b43786f95e68e052623e8aef0d4b4c0f).
+
+### User Stories
+
+#### Story 1
+
+As an Infrastructure Admin, I want to be able to deploy a 4 nodes cluster as a single operation the
+same way I deploy a full cluster. Aware of the risks and downsides, I want to be able to still
+pursue this deployment as it fits best the hardware I have available.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The current HTTP API allows to push full manifests. That is, a manifest that can be fed into `oc
+apply -f`. As part of this work, we need to have a way to add partial manifests and, for this, we
+need to decide whether we want to accept yaml-patch files or just regular yaml files. Here are a
+couple of examples:
+
+Partially populated manifest to overwrite the `status.infrastructureTopology` field:
+
+```yaml
+status:
+  infrastructureTopology: HighlyAvailable
+```
+
+yaml-patch version for the above:
+
+```yaml
+- op: add
+  path: /status/infrastructureTopology
+  value: HighlyAvailable
+```
+
+In the first case, the current API can be re-used. Assisted service would be changed to not overrite
+files but rather merge the manifests. This is how [this PoC was
+implemented](https://github.com/flaper87/assisted-service/commit/c6d72827b43786f95e68e052623e8aef0d4b4c0f).
+In the `yaml-patch` case, we may want to make the API accept both, full manifest, and a `yaml-patch`
+version. The API would have to allow for the user to specify, explicitly, whether the file submitted
+is a `yaml-patch` or not. Note that all this would also have to be exposed through the KubeAPI.
+
+
+### Risks and Mitigations
+
+The 3+1 topology is not part of OCP's test matrix. This, to some extent, makes it a
+non-supported topology. However, it has been accepted as a valid topology for the ZTPFW use-case.
+
+The reason for the above is that a 4-nodes cluster that uses masters for infrastructure workload
+would be sending infrastructure traffic to the master nodes. This, depending on the usage of the
+cluster, may result in master nodes being overloaded and, in some cases, even unavailable if the
+load is too high.
+
+Using a 4-node cluster must be a concious decision and it requires an understanding of what
+workloads will be run on the cluster. This topology is not much different to a compact cluster where
+there are only 3 masters. By exposing these options through the KubeAPI we will be allowing users to
+create such topologies.
+
+Finally, by exposing these options, we would also allow users to overwrite any manifest through the
```

Comment:
This is a good point, but this is not a new risk - we already give users the ability to overwrite / add complete manifests. If anything, patching support will make it more surgical with a smaller blast radius

Reply:
Alright, so please add a small remark that emphasizes that this is not a new risk